### DISCUSSION TITLE
Exclude words / blocked terms per index

### DISCUSSION THREAD

**Discussion by seriousm4x** [meilisearch/product#678] [Feedback & Feature Proposal]
Hello,

I already asked this question in [meilisearch/discussions](https://github.com/meilisearch/meilisearch/discussions/3839#discussioncomment-6240102), but was then referred here.

### Problem
You can't exclude words from search results so that multiple spellings are considered.

### Feature request
The possibility to create a list of "blocked terms" / "bad words" per index, which will be excluded from the search results.

### Example
I have several variations of the word "nazi". Lower case, capitalised, plural and variations with other words combined. The example is only one of many I'd like to exclude. It is almost impossible to enter all combinations in the stop words list. 

### Current state of filtering words in meili
There are [stop words](https://www.meilisearch.com/docs/reference/api/settings#stop-words) which exclude only exact matches. For example, if the word is `badword`, only the exact spelling will be excluded, but not `Badword`. It is almost impossible to enter all combinations in the stop words list.

### My use case
I have a VOD archive for a Twitch streamer ([wubbl0rz.tv](https://wubbl0rz.tv/)) where you can search all livestreams using voice to text transcripts. I transcode each stream to text using OpenAI Whisper and save the transcripts in Meilisearch (You can see all transcripts [here](https://github.com/AgileProggers/archiv-transcribe/tree/master/transcripts)). Now I want to be able to exclude certain words that are not shown in the results.

Of course, it would be better if these words didn't end up in Meilisearch in the first place. However, this is not possible, because I have no influence on what the streamer says.

**Comment by macraig** [comment]
Hi @seriousm4x  , thanks for taking the time to leave your feedback. 

Currently, stop words are case-sensitive due to the tokenization process. The categorization of a token (into a word, a stop word, or a separator) is made before the normalization of the token (lowercasing, diacritic removal, etc.). This means that the token we compare with the stop words list is not lowercased and doesn’t match if the casing is different.

Your use case makes sense, it's not sustainable to enter all combinations in the stop words list. We could partially solve it by inverting the current process (normalizing stop words before categorizing). This way stop words would not be case-sensitive and you could at least remove those variations. Would that work for you?

**Reply by seriousm4x** [reply]
That'd make it a lot easier!
I don't know exactly how Meili works, but is it also possible to match the words to a dictionary? For example, `cats` and `cat` would be matched to the same word. I only know that Postgres can do this with `to_tsquery`, but don't know if this is possible in Meili.

**Reply by macraig** [reply]
Hi @seriousm4x , apologies for the late reply.

> but is it also possible to match the words to a dictionary?

You can now create your own [user dictionary](https://www.meilisearch.com/docs/reference/api/settings#dictionary) and add variations of the words to it. It still doesn't solve the problem of having to add every combination manually, but it might be helpful for the most common misspellings and patterns.
